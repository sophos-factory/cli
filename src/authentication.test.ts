import { execute } from '../tests/helpers/execute';
import { withAddress } from '../tests/helpers/options';

describe('authentication', () => {
  test('rejects if token is not provided', async () => {
    await expect(
      // @ts-expect-error: We intentionally providing undefined as token here.
      execute(withAddress(['list', 'projects']), { token: undefined })
    ).rejects.toMatchSnapshot();
  });

  test('accepts if token is provided with --auth-token', async () => {
    await expect(
      execute(withAddress(['list', 'projects', '--auth-token', 'tokenabc']), {
        // @ts-expect-error: We intentionally providing undefined as token here.
        token: undefined
      })
    ).resolves;
  });
});
