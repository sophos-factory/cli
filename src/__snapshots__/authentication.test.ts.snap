// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authentication rejects if token is not provided 1`] = `
"factoryctl projects [options]

Options:
      --version          Show version number  [boolean]
  -v, --verbose          Print detailed output  [boolean]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
      --filter           Filter output using JsonPath  [string]
      --address          Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Sophos Factory Auth API server  [string] [default: FACTORY_AUTH_ADDRESS environment variable if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the projects  [string]
      --sort             Sort order  [string] [choices: "modified_asc", "modified_desc", "name_asc", "name_desc"]
      --limit  [number]
      --offset  [number]

Authentication token must be specified!
"
`;
